From: Gerald Pfeifer <gerald@pfeifer.com>
Subject: [PATCH] nsiproxy.sys: Only use SIN_ROUTER when available
Message-Id: <20210811235318.5315A33E83@hamza.pair.com>
Date: Thu, 12 Aug 2021 01:53:15 +0200 (CEST)

FreeBSD (and others) fall into the non-Linux case that is guarded by
HAVE_SYS_SYSCTL_H. Alas not all such systems feature SIN_ROUTER which
is used there, so tighten the guard.

Signed-off-by: Gerald Pfeifer <gerald@pfeifer.com>
---
 dlls/nsiproxy.sys/ip.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/nsiproxy.sys/ip.c b/dlls/nsiproxy.sys/ip.c
index 19272861cef..04f5ff3fda5 100644
--- a/dlls/nsiproxy.sys/ip.c
+++ b/dlls/nsiproxy.sys/ip.c
@@ -649,7 +649,7 @@ static NTSTATUS ipv4_neighbour_enumerate_all( void *key_data, DWORD key_size, vo
         }
         fclose( fp );
     }
-#elif defined(HAVE_SYS_SYSCTL_H)
+#elif defined(HAVE_SYS_SYSCTL_H) && defined(SIN_ROUTER)
     {
         int mib[] = { CTL_NET, PF_ROUTE, 0, AF_INET, NET_RT_FLAGS, RTF_LLINFO }, sinarp_len;
         size_t needed;

-- 
2.31.1

